tripy/tripy/frontend/tensor.py

```diff
@@ -150,7 +151,7 @@ def raw_init(
                 data = memref.create_memref_view(data)
             Storage.build_internal([], [instance.trace_tensor], data)
         else:
-            Storage.build_internal([], [instance.trace_tensor], data, dtype, device)
+            Storage.build_internal([], [instance.trace_tensor], data, None, device)
```

Comment:
Is there ever a case where we actually want to set the datatype in `Storage`? If not I think we can drop the argument from the constructor.

Reply:
The only case is for empty tensor where we cannot infer dtype from list elements. Is there a better solution for it?

Reply:
Oh empty tensor can be casted as well, I can remove it then.

Reply:
Can we default it to FP32 and then rely on our casting? That's effectively what's happening here when you pass `None`